### DISCUSSION TITLE
[Feature] Add the possibility to associate user-friendly names to IDs to quickly choose what photos to show via URL parameter

### DISCUSSION THREAD

**Discussion by Trainax** [damongolding/immich-kiosk#90] [Feature request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hello everyone. I recently started using Immich-Kiosk and I really like it. One feature I would really like to be implemented is a way to associate a user-friendly name to an ID. Let me make an example: let's say I want to show the pictures from the "Family" album. To do that I would open a broser and type `http://{immich-kiosk-URL}?album=FAMILY_ALBUM_ID`, however it's not easy to remember the ID of every album I've created, so I would need to use bookmarks (or another similar solution).

What I propose is the ability to create a separate file (or some parameters in the YAML file) to associate a user-friendly name to an ID, so that selecting what photos to show would be easier. Something like this:

```
- Album_ID:
    FAMILY_ALBUM_ID:"family"
    FRIENDS_ALBUM_ID:"friends"
    ...
- People_ID:
    GRANDMA_PERSON_ID:"grandma"
    DAD_PERSON_ID:"dad"
    ...
```

So that if I want to show photos from the "Family" album I could simply type `http://{immich-kiosk-URL}?album=family` or, similarly, if I want to show photos with my grandma I could type `http://{immich-kiosk-URL}?person=grandma`.

Do you think it could be a useful feature? Thank you very much!

**Comment by damongolding** [comment]
I do like the idea but I do have two concerns.

- How would these be set in ENV or would we enforce these to the  yaml (nothing else does this)

- They could be quite prone to user error, while `ALBUM_ID: Daniel’s crazy party?!?!🎉` is valid yaml but would break the URL. I know it’s an extreme example but not out of realm of possibility.

**Reply by Trainax** [reply]
- It could be set in ENV, I don't see a reason why it should be forced in the YAML (mine was an example)
- Maybe the documentation could specify that only lowercase ASCII strings (letters and numbers with no spaces) will be accepted by the program so the users will know how to write their user-friendly names for the IDs
